http-clients/src/test/java/org/triplea/http/client/web/socket/WebSocketConnectionTest.java

```diff
@@ -127,21 +180,13 @@ private void verifyPingerIsStarted() {
           is(true));
     }
 
-    private void givenWebSocketConnects(final boolean connects) throws Exception {
-      when(webSocketClient.connectBlocking(
-              WebSocketConnection.DEFAULT_CONNECT_TIMEOUT_MILLIS, TimeUnit.MILLISECONDS))
-          .thenReturn(connects);
-    }
-
     @Test
     @DisplayName("Verify connect failing invokes error handler and pinger is not running")
-    void connectionFailure() throws Exception {
-      givenWebSocketConnects(false);
-
-      final boolean connected =
-          webSocketConnection.connect(webSocketConnectionListener, errorHandler).get();
+    void connectionFailure() {
```

Comment:
I'm seeing this test fail when run from IDE:
```
org.opentest4j.AssertionFailedError: Expected java.util.concurrent.ExecutionException to be thrown, but nothing was thrown.

	at org.junit.jupiter.api.AssertThrows.assertThrows(AssertThrows.java:71)
	at org.junit.jupiter.api.AssertThrows.assertThrows(AssertThrows.java:37)
	at org.junit.jupiter.api.Assertions.assertThrows(Assertions.java:2952)
	at org.triplea.http.client.web.socket.WebSocketConnectionTest$SendMessageAndConnect.connectionFailure(WebSocketConnectionTest.java:186)
```
I think I'd quibble a bit more with the test case otherwise. The `@DisplayName` message is not really accurate anymore. It's also very notable that we can/need to rely on exception handling.

It's also notable as well that the method does not document the `ExecutionException` as being a possibility for somethign to be thrown:
```
   * @throws IllegalStateException Thrown if connection is already open (eg: connect called twice).
   * @throws IllegalStateException Thrown if connection has been closed (ie: 'close()' was called)
   */
```

Third, we pass an error handler to the method, yet we also have an exception that is possible to be thrown. Part of the purpose of the error handler is to invoke that instead of throwing an exception.

The local IDE failure does need to be sorted out, we can't have tests being non-deterministic and breaking when run from one context and not another. I suspect getting back to a boolean returning only method and not allowing an exception to be thrown, returning false and invoking the error handler would be more inline with the original method contract.

Reply:
For me the test case works fine, even when running it 10000 times.
The test basically checks if connecting to `wss://server.invalid` fails, which should always be the case because `.invalid` is a reserved invalid TLD unless you changed something in your host file.
The `ExecutionException` is not thrown by the method itself, it's thrown by `Future#get` to wrap the underlying IOException that is thrown inside the future. I update the documentation to clarify the future behaviour.